subscription_events.go

```diff
@@ -0,0 +1,84 @@
+package chartmogul
+
+const subscriptionEventsEndpoint = "subscription_events"
+
+type SubscriptionEvent struct {
+	ID                        uint32      `json:"id,omitempty"`
+	DataSourceUUID            string      `json:"data_source_uuid,omitempty"`
+	CustomerExternalID        string      `json:"customer_external_id"`
+	SubscriptionSetExternalID string      `json:"subscription_set_external_id,omitempty"`
+	SubscriptionExternalID    string      `json:"subscription_external_id,omitempty"`
+	PlanExternalID            string      `json:"plan_external_id,omitempty"`
+	EventDate                 string      `json:"event_date"`
+	EffectiveDate             string      `json:"effective_date"`
+	EventType                 string      `json:"event_type"`
+	ExternalID                string      `json:"external_id,omitempty"`
+	Errors                    interface{} `json:"errors,omitempty"`
+	CreatedAt                 string      `json:"created_at,omitempty"`
+	UpdatedAt                 string      `json:"updated_at,omitempty"`
+	Quantity                  int32       `json:"quantity,omitempty"`
+	Currency                  string      `json:"currency,omitempty"`
+	AmountInCents             int32       `json:"amount_in_cents,omitempty"`
+	TaxAmountInCents          int32       `json:"tax_amount_in_cents,omitempty"`
+	RetractedEventId          string      `json:"retracted_event_id,omitempty"`
+}
+
+type SubscriptionEvents struct {
+	SubscriptionEvents []SubscriptionEvent `json:"subscription_events"`
```

Comment:
This should be of type `[]*SubscriptionEvent`

Reply:
Putting complex types as _values_ into arrays (slices) can lead to various problems. It's a while since I touched Go, so I don't remember exactly, but arrays of pointers align more closely with OOP languages and I remember we had some issues with interfaces and copying slices.
Well, if nothing else, it's a convention in this library.

We cannot just change the type on an old struct however, because that would be breaking change.